firestore/src/include/firebase/firestore/query.h

```diff
@@ -637,6 +637,18 @@ class Query {
       std::function<void(const QuerySnapshot&, Error, const std::string&)>
           callback);
 
+  /**
+   * @brief Returns true if this `Query` is valid, false if it is not valid. An
```

Comment:
I am not sure if we need to mention this, but the main usage for this API will likely be via `CollectionReference.is_valid`. I am not sure if all of our users are aware of our type hierarchy.

Reply:
Yeah, I also do not know of a good way to address this.